resampler/src/speex.rs

```diff
@@ -334,14 +333,14 @@ impl SpeexResamplerState {
                 olen -= omagic
             }
             if 0 == self.magic_samples[channel_index as usize] {
-                j = 0u32;
-                while j < ichunk {
-                    self.mem[mem_idx
-                        + j as usize
-                        + (self.filt_len - 1) as usize] =
-                        in_0[(j * istride_save) as usize] as f32;
-                    j += 1
-                }
+                self.mem
+                    .iter_mut()
+                    .skip(mem_idx + (self.filt_len - 1) as usize)
+                    .enumerate()
+                    .take(ichunk as usize)
+                    .for_each(|(j, x)| {
+                        *x = in_0[j * (istride_save as usize)] as f32;
```

Comment:
We can use `.zip(in_0.iter())` here and remove the `j` variable definitely. We can also remove the `enumerate()`